packages/core/src/amazonqFeatureDev/session/sessionState.ts

```diff
@@ -453,24 +482,27 @@ export class MockCodeGenState implements SessionState {
 }
 
 export class PrepareCodeGenState implements SessionState {
-    public tokenSource: vscode.CancellationTokenSource
     public readonly phase = DevPhase.CODEGEN
     public uploadId: string
     public conversationId: string
+    public tokenSource: vscode.CancellationTokenSource
     constructor(
         private config: SessionStateConfig,
         public filePaths: NewFileInfo[],
         public deletedFiles: DeletedFileInfo[],
         public references: CodeReference[],
         public tabID: string,
-        private currentIteration: number,
+        public currentIteration: number,
         public codeGenerationRemainingIterationCount?: number,
         public codeGenerationTotalIterationCount?: number,
         public uploadHistory: UploadHistory = {},
+        public superTokenSource?: vscode.CancellationTokenSource,
```

Comment:
what does superTokenSource mean? also instead of falling back to a default you can just do:
```
superTokenSource: vscode.CancellationTokenSource = new vscode.CancellationTokenSource()
```